authz/rbac_translator.go

```diff
@@ -266,6 +281,48 @@ func parseRules(rules []rule, prefixName string) (map[string]*v3rbacpb.Policy, e
 	return policies, nil
 }
 
+func parseAuditLoggingOptions(options auditLoggingOptions) (*v3rbacpb.RBAC_AuditLoggingOptions, error) {
+	optionsRbac := v3rbacpb.RBAC_AuditLoggingOptions{}
+	if options.AuditCondition == "" && len(options.AuditLoggers) == 0 {
+		// No audit logger parsed - fine
+		return &optionsRbac, nil
+	} else if options.AuditCondition == "" {
+		return &optionsRbac, fmt.Errorf("AuditLogger config present but missing AuditCondition")
+	} else if len(options.AuditLoggers) == 0 {
+		return &optionsRbac, fmt.Errorf("AuditCondition present but missing AuditLogger config")
+	}
+
+	switch options.AuditCondition {
+	case "NONE":
+		optionsRbac.AuditCondition = v3rbacpb.RBAC_AuditLoggingOptions_NONE
+	case "ON_DENY":
+		optionsRbac.AuditCondition = v3rbacpb.RBAC_AuditLoggingOptions_ON_DENY
+	case "ON_ALLOW":
+		optionsRbac.AuditCondition = v3rbacpb.RBAC_AuditLoggingOptions_ON_ALLOW
+	case "ON_DENY_AND_ALLOW":
+		optionsRbac.AuditCondition = v3rbacpb.RBAC_AuditLoggingOptions_ON_DENY_AND_ALLOW
+	default:
+		return &optionsRbac, fmt.Errorf("Failed to parse AuditCondition %v. Allowed values {NONE, ON_DENY, ON_ALLOW, ON_DENY_AND_ALLOW}", options.AuditCondition)
```

Comment:
Now that you are returning a pointer, please return nil when there is an error. That's the idiomatic pattern in Go.

Reply:
Fixed